### PR TITLE
better dns discovery error and doc

### PR DESCRIPTION
**Pull request by xiang90**
1. move dns discovery cluster checking to etcdmain and give it a better error message

2. add doc around it

Fix https://github.com/coreos/etcd/issues/2426

/cc @robszumski @kelseyhightower

### CODE REVIEW

File: Documentation/clustering.md [hunk 1]
```diff
@@ -300,6 +300,8 @@ infra2.example.com.	300	IN	A	10.0.1.12
 
 etcd cluster members can listen on domain names or IP address, the bootstrap process will resolve DNS A records.
 
+Please make sure the resolved address in `-initial-advertise-peer-urls` matches the one of the resolved addresses in the SRV targets. That is how an etcd server finds out if it belongs to the cluster defined in the SRV records.
```
Comment (robszumski) [inline Documentation/clustering.md]: You can remove the please:

"**The** resolved address in `-initial-advertise-peer-urls` **must match** one of"

Comment (robszumski) [inline Documentation/clustering.md]: "**The** etcd server **reads the resolved address to find** out if it belongs to the cluster defined in the SRV records."

**Comment by xiang90** [comment]
@robszumski Fixed. Can you please take another look?

**Comment by robszumski** [comment]
LGTM

File: Documentation/clustering.md [hunk 2]
```diff
@@ -300,6 +300,8 @@ infra2.example.com.	300	IN	A	10.0.1.12
 
 etcd cluster members can listen on domain names or IP address, the bootstrap process will resolve DNS A records.
 
+The resolved address in `-initial-advertise-peer-urls` *must match* one of the resolved addresses in the SRV targets. The etcd member reads the resolved address to find out if it belongs to the cluster defined in the SRV records
```
Comment (yichengq) [inline Documentation/clustering.md]: it miss a mark

**Comment by yichengq** [comment]
lgtm

**Pull request merged**